src/Curation/Curation.router.spec.ts

```diff
@@ -92,230 +122,420 @@ describe('when handling a request', () => {
+        })
+
+        it('should reject with invalid schema message', async () => {
+          await expect(router.updateItemCuration(req)).rejects.toThrow(
+            'Invalid schema'
+          )
+        })
       })
     })
 
     describe('when the curation does not exist', () => {
-      it('should reject with curation not found message', async () => {
-        mockHasAccessToCollection.mockResolvedValueOnce(true)
-
-        jest
-          .spyOn(Curation, 'getLatestForCollection')
-          .mockResolvedValueOnce(undefined)
+      let req: AuthRequest
 
-        const req = {
+      beforeEach(() => {
+        req = {
           auth: { ethAddress: 'ethAddress' },
-          params: { collectionId: 'collectionId' },
+          params: { id: 'some id' },
           body: {
             curation: {
               status: 'rejected',
             },
           },
         } as any
+      })
 
-        await expect(router.updateCuration(req)).rejects.toThrow(
-          'Curation does not exist'
-        )
+      describe('when updating a collection', () => {
+        beforeEach(() => {
+          service = mockServiceWithAccess(CollectionCuration, true)
+          jest.spyOn(service, 'getLatestById').mockResolvedValueOnce(undefined)
+        })
+
+        it('should reject with curation not found message', async () => {
+          await expect(router.updateCollectionCuration(req)).rejects.toThrow(
+            'Curation does not exist'
+          )
+        })
       })
-    })
 
-    describe('when everything is fine', () => {
-      it('should resolve with the updated curation', async () => {
-        mockHasAccessToCollection.mockResolvedValueOnce(true)
+      describe('when updating an item', () => {
+        beforeEach(() => {
+          service = mockServiceWithAccess(ItemCuration, true)
+          jest.spyOn(service, 'getLatestById').mockResolvedValueOnce(undefined)
+        })
 
-        jest
-          .spyOn(Curation, 'getLatestForCollection')
-          .mockResolvedValueOnce({ id: 'curationId' } as any)
+        it('should reject with curation not found message', async () => {
+          await expect(router.updateItemCuration(req)).rejects.toThrow(
+            'Curation does not exist'
+          )
+        })
+      })
+    })
 
-        const updateSpy = jest
-          .spyOn(Curation, 'update')
-          .mockResolvedValueOnce({} as any)
+    describe('when everything is fine', () => {
+      let req: AuthRequest
 
-        const req = {
+      beforeEach(() => {
+        req = {
           auth: { ethAddress: 'ethAddress' },
-          params: { collectionId: 'collectionId' },
+          params: { id: 'some id' },
           body: {
             curation: {
               status: 'rejected',
             },
           },
         } as any
+      })
 
-        await expect(router.updateCuration(req)).resolves.toStrictEqual({})
+      describe('when updating a collection', () => {
```

Comment:
What if we add the "curation" word here.
```suggestion
      describe('when updating a collection curation', () => {
```
Or, what if we change it to:
```suggestion
      describe('when the curation to be updated belong to a collection', () => {
```
or:
```suggestion
      describe('and the curation to be updated belong to a collection', () => {
```

The same goes for the other tests that use the same text.

Reply:
I think I'm going to go with the first one for brevity, as I have to change it for items too